contentcuration/contentcuration/models.py

```diff
@@ -1235,3 +1238,14 @@ class Invitation(models.Model):
     class Meta:
         verbose_name = _("Invitation")
         verbose_name_plural = _("Invitations")
+
+
+class Task(models.Model):
+    """Asynchronous tasks"""
+    id = models.CharField(primary_key=True, max_length=50)   # Get the id from Celery task
+    task_type = models.CharField(max_length=50)
+    created = models.DateTimeField(default=timezone.now)
+    status = models.CharField(max_length=10)
+    is_progress_tracking = models.BooleanField(default=False)
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, related_name="task")
```

Comment:
I'm thinking we may want to add a ManyToManyField on channel here, so that we can run an indexed query to find out if a particular channel has tasks running on it. (e.g. for a copy operation, we would want to know to know to disable editing of the relevant source and destination nodes while the copy is happening.)

Thoughts?

Reply:
I think if we are just planning to do some simple checks such as finding out if a channel has related tasks running, then the `metadata` field (JSONField) could be efficient enough? e.g.
```
Task.objects.filter(metadata__channel_id='66a31fdbdd745c7c9e499206f3c4cced')
<QuerySet [<Task: Task object>, <Task: Task object>]>
```
One good thing about using JSONField that I could think of is that we can define different keys for the JSONFields with regard to different endpoints. For example, `channel_id` in the `metadata` for `publish_channel` endpoint, and `node_id` in the metadata for the `duplicate_nodes` endpoint. 

If we add channel as a ManyToManyField in the Task model, then as Jordan mentioned, we will probably also need contentnode as a ManyToManyField(or something like that) for `duplicate_nodes` endpoint, which may not be used in the `publish_channel` endpoint.

But if we plan to do more operations on channel/contentnode/asynctask other than just the simple checks, I agree that having the channel/contentnode as fields in the Task model would be better :)

Reply:
Sorry for the late reply, I forgot that we could query JSONField like this. In that case, yeah, I think using the metadata field solves the problem nicely!